### PR TITLE
CMake: fix build for Apple Silicon hosts

### PR DESCRIPTION
**Pull request by MaxDesiatov**
When building with `build-script` using these arguments

```
utils/build-script --skip-build-benchmarks
  --skip-ios --skip-watchos --skip-tvos
  --swift-darwin-supported-archs "arm64"
  --sccache --release-debuginfo --test
```

the build fails with

```
ninja: error: 'stdlib/swift-test-stdlib-macosx-x86_64',
needed by 'stdlib/CMakeFiles/swift-test-stdlib', missing and no known rule to make it
```

I think that the "Getting Started" guide should avoid hardcoding `x86_64` arguments, and suggest using `$(uname -m)` instead. `SWIFT_PRIMARY_VARIANT_ARCH_default` could also get its value from `CMAKE_HOST_SYSTEM_PROCESSOR` in the root `CMakeLists.txt`.

Resolves SR-13943.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Comment by stephentyrone** [comment]
Tagging @compnerd and @gottesmm, since they're my usual go-tos for CMake-fu.

**Review by compnerd — APPROVED** [review]
Id want an okay from Michael on this as well; the CMake side of the change is proper, the question is, is there anything that depends on this?

**Comment by varungandhi-apple** [comment]
Thanks for the PR. There are some other places where `macosx-x86_64` is specified directly and you may want to update those. Specifically:

1. `utils/coverage/coverage-generate-data`
2. `benchmark/README.md`
3. `clang-tidy` instructions in `docs/DebuggingTheCompiler.md`.

**Comment by MaxDesiatov** [comment]
@varungandhi-apple thanks for highlighting those, update now.

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Review by varungandhi-apple — APPROVED** [review]

**Comment by varungandhi-apple** [comment]
@MaxDesiatov can this be merged/are you waiting on something specific?

**Comment by MaxDesiatov** [comment]
IDK, do I need for an explicit approval from @gottesmm before this can be merged? As @compnerd mentioned above:

> Id want an okay from Michael on this as well

**Comment by davidungar** [comment]
@MaxDesiatov @compnerd @varungandhi-apple I independently stumbled across this issue over break. (Too bad this PR wasn't merged then.) My version of a fix is in https://github.com/apple/swift/pull/35231 . 
It uses `SWIFT_HOST_VARIANT_ARCH` instead of the fix in this PR. Which PR would be better to merge?

**Comment by varungandhi-apple** [comment]
IMO we should merge this, since it also updates the documentation. We don't need to be blocked on review, we can change things after-the-fact if there's an issue.

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test macOS platform

**Pull request merged**
